spec/services/make_offer_spec.rb

```diff
@@ -11,60 +16,27 @@
-        expect {
-          make_offer.save!
-        }.to raise_error(
-          ProviderAuthorisation::NotAuthorisedError,
-          'You are not permitted to view this application. The specified course is not associated with any of your organisations. You do not have the required user level permissions to make decisions on applications for this provider.',
-        )
-      end
-    end
+      expect { make_offer.save! }.not_to raise_error
 
-    describe 'if the application choice cannot transition to the offer state' do
-      let(:application_choice) { create(:application_choice, status: :pending_conditions) }
-      let(:provider_user) do
-        create(:provider_user,
-               :with_make_decisions,
-               providers: [application_choice.course_option.provider])
-      end
+      expect(MakeAnOffer).to have_received(:new).with(actor: provider_user,
+                                                      application_choice: application_choice,
+                                                      course_option: course_option,
+                                                      offer_conditions: conditions)
 
-      it 'throws an exception' do
-        expect {
-          make_offer.save!
-        }.to raise_error(Workflow::NoTransitionAllowed, 'There is no event make_offer defined for the pending_conditions state')
-      end
+      expect(make_an_offer).to have_received(:save)
     end
 
-    describe 'if the provided details are correct' do
-      let(:application_choice) { create(:application_choice, status: :awaiting_provider_decision) }
-      let(:provider_user) do
-        create(:provider_user,
-               :with_make_decisions,
-               providers: [application_choice.course_option.provider])
-      end
-
-      it 'then it executes the service without errors ' do
-        set_declined_by_default = instance_double(SetDeclineByDefault, call: true)
-        send_new_offer_email_to_candidate = instance_double(SendNewOfferEmailToCandidate, call: true)
-        allow(SetDeclineByDefault)
-          .to receive(:new).with(application_form: application_choice.application_form)
-          .and_return(set_declined_by_default)
-        allow(SendNewOfferEmailToCandidate)
-          .to receive(:new).with(application_choice: application_choice)
-          .and_return(send_new_offer_email_to_candidate)
-
-        make_offer.save!
+    it 'raises an error when the save is unsuccessful' do
+      make_an_offer = instance_double(MakeAnOffer)
+      allow(make_an_offer).to receive(:save).and_return(false)
+      allow(MakeAnOffer).to receive(:new) { make_an_offer }
 
-        expect(SetDeclineByDefault).to have_received(:new).with(application_form: application_choice.application_form)
-        expect(SendNewOfferEmailToCandidate).to have_received(:new).with(application_choice: application_choice)
-      end
+      expect { make_offer.save! }.to raise_error('Unable to complete save on make_an_offer')
```

Comment:
The purpose of this spec is to verify that the correct set of services is called. As the goal is to eventually transition to only using the MakeOffer service, which should ensure that the correct set of services are called, these changes are not required.

Reply:
The spec for `MakeAnOffer` already verifies these services are called and there is another scenario in the `MakeOffer` spec which verifies that `MakeAnOffer` is called. If we want a spec that verifies the full range of actions `MakeOffer` performs, we have a comprehensive one included in the previous PR. With this PR's approach, the only checks that feel appropriate are the ones proposed.